pkg/runtime/interfaces.go

```diff
@@ -30,12 +30,23 @@ const (
 	APIVersionInternal = "__internal"
 )
 
+// GroupVersioner conveys information about a desired target version for objects being converted.
+type GroupVersioner interface {
```

Comment:
I guess I would expect this interface to provide just one function, which takes in a fully specified GVK and returns a fully specified GVK. It answers the question "what should this type be converted to?"